src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -909,15 +922,26 @@ internal virtual void PrepareSession()
                     // UseDefaultCredentials will overwrite the supplied credentials.
                     webProxy.UseDefaultCredentials = true;
                 }
-
-                WebSession.Proxy = webProxy;
+                // We don't want to update the WebSession unless the proxies are different
+                // as that will require us to create a new HttpClientHandler and lose connection
+                // persistence
+                if (!webProxy.Equals(WebSession.Proxy))
+                {
+                    WebSession.Proxy = webProxy;
+                }
+            }
+            if (MyInvocation.BoundParameters.ContainsKey("SslProtocol"))
```

Comment:
I don't think we need to check for SslProtocol
```cs
WebSession.SslProtocol = SslProtocol
```
should be ok

Reply:
SslProtocol is a struct type so if it's not provided on the command line it will be the default value. That would mean we'd require the user to always supply the SslProtocol on every IWR if they chose a non-default value unless we check for it explicitly being present before deciding whether to use it to 'change' the webrequestsession, which would then cause us to open a new request.

The logic is if the user doesn't supply the parameter at all, but they've provided a Session, we should use whatever was set in the Session, not go back to the default.